### PR TITLE
Add README

### PR DESCRIPTION
**Pull request by yashtewari**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,67 @@
-# opa-rails-middleware
\ No newline at end of file
+# opa-rails-middleware
+
+<p align="center"><img src="img/logo-build.png" class="center" alt="build-logo" width="30%"/></p>
```
Comment (Ofir-Lapid) [inline README.md:3]: Need to fix the images

File: README.md [hunk 2]
```diff
@@ -1 +1,67 @@
...
+
+ 1. `hostname` - The hostname of the Policy Decision Point (PDP) **Default is localhost**
+ 2. `port` - The port at which the PDP service is serving authz decisions **Default is 8181**
+ 3. `policy_path` - Full path to the policy (including the rule) that decides whether requests should be authorized **Default is /authz/allow**
+ 4. `retry_max_attempts` - the maximum number of retry attempts in case a failure occurs. **Default is 2**.
+ 5. `read_timeout_milliseconds` - Read timeout for requests in milliseconds. **Default is 5000**
+ 6. `connection_timeout_milliseconds` - Connection timeout in milliseconds. **Default is 5000**
+ 7. `retry_backoff_milliseconds` - The number of milliseconds to wait between two consecutive retry attempts. **Default is 250**
+
+[How to get your pdp's hostname and port?](https://docs.build.security/policy-decision-points-pdp#pdp-instances-section)
```
Comment (Ofir-Lapid) [inline README.md:67]: Please add a request and a response example

File: README.md [hunk 3]
```diff
@@ -1 +1,67 @@
...
+
+You can see how this is done [in the example](/example/config/application.rb).
+   
+ ### Configuration
+
+ 1. `hostname` - The hostname of the Policy Decision Point (PDP) **Default is localhost**
+ 2. `port` - The port at which the PDP service is serving authz decisions **Default is 8181**
+ 3. `policy_path` - Full path to the policy (including the rule) that decides whether requests should be authorized **Default is /authz/allow**
+ 4. `retry_max_attempts` - the maximum number of retry attempts in case a failure occurs. **Default is 2**.
+ 5. `read_timeout_milliseconds` - Read timeout for requests in milliseconds. **Default is 5000**
```
Comment (Ofir-Lapid) [inline README.md:63]: I saw that in other middlewares the default timeout is 1000 ms.. what is the default according to the doc ?

File: README.md [hunk 4]
```diff
@@ -1 +1,67 @@
-# opa-rails-middleware
\ No newline at end of file
+# opa-rails-middleware
+
+<p align="center"><img src="img/logo-build.png" class="center" alt="build-logo" width="30%"/></p>
+
+## Abstract
+[build.security](https://docs.build.security/) provides simple development and management for your organization's authorization policy.
```
Comment (Ofir-Lapid) [inline README.md:6]: Not critical and not a "must"
But why did you change the format ? 

For example - symfony 
build.security provides simple development and management for your organization's authorization policy. opa-symfony-middleware is a PHP Symfony middleware intended for performing authorization requests against build.security PDP(Policy Decision Point)/OPA.

This package is built for PHP v8.0 and above and Symfony v4.22 and above.

**Review by Ofir-Lapid — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (yashtewari) [reply README.md:3]: The images are working!

File: README.md (hunk 2 above)
Comment (yashtewari) [reply README.md:67]: I don't exactly understand. Do you mean an example of what the PDP input looks like?

File: README.md (hunk 3 above)
Comment (yashtewari) [reply README.md:63]: I think it's 5000 in most of them, which is definitely too much. Let's make a task and change all the defaults in one go. Until then it's fine to leave it as in in my opinion.

File: README.md (hunk 2 above)
Comment (Ofir-Lapid) [reply README.md:67]: https://github.com/build-security/opa-symfony-middleware

Check the last part of the repo, you would see here a request and a response example

File: README.md (hunk 3 above)
Comment (Ofir-Lapid) [reply README.md:63]: opa sympony - 1000
.net - 1000
opa-java-sprint -5000
opa java spring 5000...

:/ 

Ok so leave it as it is

File: README.md (hunk 2 above)
Comment (yashtewari) [reply README.md]: Added!

File: README.md [hunk 5]
```diff
@@ -1 +1,114 @@
...
+```
+gem "policy_decision_point"
+```
+
+and run `bundle install`
+
+---
+### Simple usage
+
+To intercept and authorize requests being made to your Rails application, add the following line to your `config/application.rb` file, under the `Application` class:
```
Comment (Ofir-Lapid) [inline README.md:41]: IMO, I would rephrase it a bit to ease the reading.
Something like - 

Simple usage
To intercept.....with default values.
Add the following line....

Advanced usage
...

**Review by Ofir-Lapid — APPROVED** [review]
